src/OpenTelemetry.Api/Trace/TracerProvider.cs

```diff
@@ -36,20 +36,13 @@ protected TracerProvider()
         /// </summary>
         public static TracerProvider Default { get; } = new TracerProvider();
 
-        /// <inheritdoc/>
-        public void Dispose()
-        {
-            this.Dispose(true);
-            GC.SuppressFinalize(this);
-        }
-
         /// <summary>
         /// Gets a tracer with given name and version.
         /// </summary>
         /// <param name="name">Name identifying the instrumentation library.</param>
         /// <param name="version">Version of the instrumentation library.</param>
         /// <returns>Tracer instance.</returns>
-        public Tracer GetTracer(string name, string version = null)
+        public static Tracer GetTracer(string name, string version = null)
```

Comment:
I think we should tweak this design.

* `public class TracerProvider` -> `public static class TracerProvider`, remove IDisposable.
* Change `GetTracer` from `return new Tracer(new ActivitySource(name, version));` to `return new Tracer(name, version));`
* Change `Tracer` so that it does the `new ActivitySource` in ctor, add IDisposable, dispose the ActivitySource it creates in Dispose.

As it is right now, every call to GetTracer creates an ActivitySource which is never disposed. What the above is doing is essentially moving that ActivitySource management to Tracer.

Reply:
That's a bit more complicated...if we move from `public class` to `public static class`, we would have to refactor the entire project, because `TracerProvider` is used as base from `TracerProviderSdk` and in our extension methods. We will have to think carefully

Reply:
Lets address it separately. Can you revert changes to this file to proceed with this PR of FxCop changes.

Reply:
just reverted the `TracerProvider` changes.